torch/fx/passes/tensorify_python_scalars.py

```diff
@@ -0,0 +1,245 @@
+                    (expr_to_sym_proxy[expr].node,),
+                ),
+                tracer=tracer,
+            )
+
+        # cache constants, why not
+        if isinstance(expr, (Integer, Number, BooleanAtom)):
+            dtype = None
+            c: Union[bool, int, float]
+            if isinstance(expr, BooleanAtom):
+                dtype = torch.bool
+                c = bool(expr)
+            elif isinstance(expr, sympy.Integer):
+                dtype = torch.int64
+                c = int(expr)
+            elif isinstance(expr, sympy.Number):
+                dtype = torch.float64
+                c = float(expr)
+
+            expr_to_tensor_proxy[expr] = fx.Proxy(
+                graph.call_function(
+                    torch.ops.aten.scalar_tensor.default, (c,), {"dtype": dtype}
+                ),
+                tracer=tracer,
+            )
+
+        if expr in expr_to_tensor_proxy:
+            return expr_to_tensor_proxy[expr]
+
+        # don't cache
+        if isinstance(expr, Symbol):
+            return sympy_interp(Analysis, expr_to_tensor_proxy, expr)  # type: ignore[arg-type]
+
+        # hash cons on arguments, run expr handler
+        expr_to_tensor_proxy[expr] = _run_sympy_handler(
+            Analysis,
+            [_sympy_interp(arg) for arg in expr.args],  # type: ignore[arg-type]
+            expr,
+        )
+
+        return expr_to_tensor_proxy[expr]
+
+    nodes = list(graph.nodes)
+    for i, node in enumerate(nodes[:-1]):
+        with graph.inserting_before(
+            nodes[i + 1] if node not in placeholders else first_non_placeholder
+        ):
+            # Look for tensor.item() calls on placeholders
+            if unbacked_bindings := node.meta.get("unbacked_bindings"):
+                for s, keypath in unbacked_bindings.items():
+
+                    def go(
+                        node: fx.Node, keypath: tuple[Any, ...]
+                    ) -> Union[fx.Node, None]:
+                        if keypath == ():
+                            return node
+                        elif (
+                            hasattr(keypath[0], "name")
+                            and keypath[0].name == "item"
+                            and isinstance(keypath[0], CallMethodKey)
+                        ):
+                            return go(
+                                graph.call_method(keypath[0].name, (node,)), keypath[1:]
+                            )
+                        else:
+                            return None
+
+                    src_node = go(node, keypath)
+                    if (
+                        src_node is not None
+                        and src_node.op == "call_function"
+                        and src_node.target
+                        is torch.ops.aten._local_scalar_dense.default
+                    ):
+                        # TODO: dtype conversion, so that we don't keep at too
+                        # low precision
```

Comment:
So, specifically, what this means is if someone passes in a f32[] scalar tensor and then immediately calls item() on it, the logic here would just take the f32[] tensor directly, which means we'd do all the subsequent tensor compute in f32[] rather than in f64[] (which is what the Python compute would have done). It's worth fixing this, even if it's as simple as preventing us from assigning tensor/sym proxies when the original tensor is f32 (this will never happen for SymFloat to tensor argument changes).